crates/cargo-contract/src/cmd/upload.rs

```diff
@@ -88,23 +88,26 @@ impl UploadCommand {
         } else {
             let upload_result = upload_exec.upload_code().await?;
             let display_events = upload_result.display_events;
-            let output = if self.output_json() {
+            let output_events = if self.output_json() {
                 display_events.to_json()?
             } else {
                 display_events.display_events(
                     self.extrinsic_cli_opts.verbosity()?,
                     upload_exec.token_metadata(),
                 )?
             };
-            println!("{output}");
```

Comment:
If upload_result.code_stored is None, we won't see any output_events. Is this intentional?

Reply:
Yeah, it is fine; events are not kept when extrinsics fail.